### PR TITLE
Jsonparser pr

### PR DESCRIPTION
**Pull request by brokenprogrammer**
Added a asyncronous JSON Parser parsing url given in the parameters and using a callback function to retrieve the JSON when the request is finished.

The JSON Parser supports checking errors: Invalid URL, Invalid JSON, No Internet Connection, Request Timeout.

### CODE REVIEW

File: src/Asserts.ts [hunk 1]
```diff
@@ -86,5 +86,40 @@ module OpenWeatherJS {
...
+
+		/**
+		 * Validates provided value is a URL and throws a new TypeError
+		 * with specified message otherwise..
+		 *
+		 * @param value   - a value being tested.
+		 * @param message - a short description of the assertion.
+		 */
+		static isUrl(value: string, message: string): void {
+			var yourRegularExpression = /(^|\s)((https?:\/\/)?[\w-]+(\.[\w-]+)+\.?(:\d+)?(\/\S*)?)/gi;
```
Comment (andreypudov) [inline src/Asserts.ts]: Can you please could it something differently to provide more information about this regular expression?

File: src/Asserts.ts [hunk 2]
```diff
@@ -86,5 +86,40 @@ module OpenWeatherJS {
...
+
+		/**
+		 * Validates provided value is a JSON Object and throws a new TypeError
+		 * with specified message otherwise..
+		 *
+		 * @param value   - a value being tested.
+		 * @param message - a short description of the assertion.
+		 */
+		static isJSONString(value: string, message: string): void {
+			try{
```
Comment (andreypudov) [inline src/Asserts.ts]: try[space]{ the same for catch, if and other statements.

File: src/Asserts.ts [hunk 3]
```diff
@@ -86,5 +86,40 @@ module OpenWeatherJS {
...
+			var matcher = yourRegularExpression;
+
+			var match = value.match(matcher);
+			if (!match){
+				throw new TypeError(message);
+			}
+		}
+
+		/**
+		 * Validates provided value is a JSON Object and throws a new TypeError
```
Comment (andreypudov) [inline src/Asserts.ts]: otherwise..? it also a raw error class.

File: src/JSONParser.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+		 * Sends a XMLHttpRequest to the given url returning the JSON
+     * response from the url. Throws a TypeError if a bad url is placed in parameters,
+     * throws Error on timeout and internet connection failure.
+		 *
+		 * @param url   - URL to send request to.
+     * @param done - Callback function letting you use the parsed object.
+		 * @return Object - JSON Object.
+		 */
+    static Parse(url: string, done: (obj: any) => void): void{
+      Asserts.isUrl (url, 'URL is invalid.');
```
Comment (andreypudov) [inline src/JSONParser.ts]: It seems like you are using different indentation between files - for and two spaces or tabs. It's better to use only spaces, where tab is equal to four spaces.

File: src/JSONParser.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+      var xmlHttp = new XMLHttpRequest ();
+
+      xmlHttp.onreadystatechange = function () {
+          if (xmlHttp.readyState == 4) {
+            try{
+              if (xmlHttp.status == 200) {
+                var obj = JSON.parse(xmlHttp.responseText);
+                console.log(obj);
```
Comment (andreypudov) [inline src/JSONParser.ts]: We don't need any console out put from library core.

File: build/OpenWeatherJS.js [hunk 6]
```diff
@@ -81,4 +100,39 @@ var OpenWeatherJS;
...
+        }
+        JSONParser.Parse = function (url, done) {
+            OpenWeatherJS.Asserts.isUrl(url, 'URL is invalid.');
+            var xmlHttp = new XMLHttpRequest();
+            xmlHttp.onreadystatechange = function () {
+                if (xmlHttp.readyState == 4) {
+                    try {
+                        if (xmlHttp.status == 200) {
+                            var obj = JSON.parse(xmlHttp.responseText);
+                            console.log(obj);
```
Comment (andreypudov) [inline build/OpenWeatherJS.js]: Please remove console output from library core.

**Comment by andreypudov** [comment]
I would like to say that you change is good. I just added some comments to minor things that better to change. Please fix them and push changes to this brunch.

File: src/Asserts.ts (hunk 1 above)
Comment (brokenprogrammer) [inline src/Asserts.ts]: Could you clairify?
Are you looking for comments?

File: src/Asserts.ts (hunk 3 above)
Comment (brokenprogrammer) [inline src/Asserts.ts]: Could you clairify?

**Comment by brokenprogrammer** [comment]
Its great to hear that you think the changes is good! 
I wanted to ask is the use of a callback function here a good choice? I found no other way to do it when making the parse asynchronous, also could you look over the unit tests? I had some difficulties implementing tests for the callback function, also i didn't know how to implement tests for all the different errors.

**Pull request merged**